### DISCUSSION TITLE
不支持wrapper类型

### DISCUSSION THREAD

**Discussion by RisingV** [starwing/lua-protobuf#175] [General]
目前发现如果protobuf的定义里包含了wrapper，如下
`
import "google/protobuf/wrappers.proto"; 
`
如果数据结构定义中使用了wrapper字段，那么通过lua-protobuf去反序列化会出现wrapper字段为空缺失的情况。进行序列化的过程也是同样的，wrapper字段会丢失。

**Comment by starwing** [comment]
这些高级特性（包括any）都是需要额外代码的，并不是pb标准的组成部分。你可以自己通过`pb.hook`接口得到自动化的支持。具体请看文档。

**Comment by rainingmaster** [comment]
可否给一些更加详细的例子，关于使用 `pb.hook` 来解析特殊字段。感谢

**Comment by starwing** [comment]
你可以看中文的官方文档（README.zh.md），如果还有问题可以在这里提问～

**Reply by cheapWhite** [reply]
这个问题最终解决了？有没有一些实际的例子

**Comment by jgrnrt** [comment]
I sadly don't understand pb.hook(). Did someone find a solution or has an example?
